wp-content/plugins/tw-rss-helper/includes/rss.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Add support to site RSS feed
+ *
+ * @since 1.0.0
+ */
+
+/**
+ * Add audio enclosure to RSS feed.
+ *
+ * This function will get post_rss_audio meta from a post.
+ * WARNING: This function will only work if ACF is active.
+ *
+ * @since 1.0.0
+ */
+function tw_rss_helper_audio_enclosure() {
+
+	// Check if ACF is active
+	if ( !function_exists('get_field') ) {
+		return false;
+	}
+
+	global $post;
+
+	// Get audio metadata.
+	$audio_id = get_post_meta( $post->ID, 'audio', true );
+
+	$audio_media = [];
+
+	$audio_url = wp_get_attachment_url( $audio_id );
+
+	// Bail early if no audio URL is found.
+	if ( $audio_url ) {
+
+		$audio_media['url'] = $audio_url;
+
+	} else {
+
+		return false;
+	}
+
+	$audio_filesize = tw_rss_helper_get_external_audio_url_filesize( $audio_url );
+
+	// If filesize is found, add it to the audio media array.
+	if ( $audio_filesize ) {
+		$audio_media['filesize'] = $audio_filesize;
+	}
+
+	// Build the default enclosure HTML.
+	$enclosure_html = sprintf(
+		'<enclosure url="%s" type="audio/mpeg" length="%s"/>',
```

Comment:
Can we also retrieve the mime type of the file instead of assuming they will always be MP3's. Some older content may be using WAV files, or we may want to change to another format in the future.

Reply:
Good point @rpeterman-gp, yes we can see if the mime type is available or use the extension to identify different types.
And then have the audio/mpeg as default in case we have an unknown one.

Reply:
@rpeterman-gp we added the code to read external file mime type if it is available, if it is not available we are going to try to use the WP method to get mimetype from extensions and finally, if it is an unknown mimetype we will return audio/mpeg by default.